src/internal/hooks/use-base-component/index.ts

```diff
@@ -26,8 +28,14 @@ export default function useBaseComponent<T = any>(
   config?: ComponentConfiguration,
   analyticsMetadata?: AnalyticsMetadata
 ) {
-  useTelemetry(componentName, config);
+  const isVisualRefresh = useVisualRefresh();
+  const theme = getVisualTheme(THEME, isVisualRefresh);
+  useComponentMetrics(componentName, { packageSource: PACKAGE_SOURCE, packageVersion: PACKAGE_VERSION, theme }, config);
   useFocusVisible();
-  const elementRef = useComponentMetadata<T>(componentName, PACKAGE_VERSION, { ...analyticsMetadata });
+  const elementRef = useComponentMetadata<T>(
+    componentName,
+    { packageName: PACKAGE_SOURCE, version: PACKAGE_VERSION, theme },
```

Comment:
There is an annoying structure difference between `useComponentMetrics` and `useComponentMetadata`. They receive same data, but in different shape. Still some refactoring needed.